src/main/java/com/ldtteam/structurize/client/gui/AbstractBlueprintManipulationWindow.java

```diff
@@ -314,46 +322,118 @@ protected void settingsClicked()
+                return;
             }
 
-            /**
-             * Inserts the elements into each row.
-             * @param index the index of the row/list element.
-             * @param rowPane the parent Pane for the row, containing the elements to update.
-             */
-            @SuppressWarnings("resource")
-            @Override
-            public void updateElement(final int index, final Pane rowPane)
+            final ValueSpec settingSpec = optional.get();
+            final String nameTKey = settingSpec.getTranslationKey();
+
+            if (label.getText() != null && label.getText().getContents() instanceof final TranslatableContents tkey && tkey.getKey().equals(nameTKey))
             {
-                rowPane.findPaneOfTypeByID("label", Text.class).setText(Component.translatable(settings.get(index).getKey()));
-                final ButtonImage buttonImage = rowPane.findPaneOfTypeByID("switch", ButtonImage.class);
-                if (settings.get(index).getValue())
-                {
-                    buttonImage.setText(Component.translatable("options.on"));
-                }
-                else
-                {
-                    buttonImage.setText(Component.translatable("options.off"));
-                }
-                buttonImage.setTextColor(ChatFormatting.BLACK.getColor());
+                // dont update when same row
+                return;
+            }
+
+            label.setText(Component.translatable(nameTKey));
+            PaneBuilders.singleLineTooltip(Component.translatable(nameTKey + AbstractConfiguration.COMMENT_SUFFIX), rowPane);
+
+            final ButtonImage buttonImage = rowPane.findPaneOfTypeByID("switch", ButtonImage.class);
+            final TextFieldVanilla inputField = rowPane.findPaneOfTypeByID("set_input", TextFieldVanilla.class);
 
+            // TODO: instanceof switch
+            if (setting.getValue(blueprintRenderSettings) instanceof final Boolean value)
+            {
+                final EitherConfig<Boolean> typedSetting = (EitherConfig<Boolean>) setting;
+
+                inputField.off();
+                buttonImage.on();
+                buttonImage.setText(Component.translatable(value ? "options.on" : "options.off"));
                 buttonImage.setHandler((button) -> {
-                    settings.get(index).setValue(!settings.get(index).getValue());
-                    Network.getNetwork().sendToServer(new SyncSettingsToServer());
-                    RenderingCache.getOrCreateBlueprintPreviewData(bluePrintId).syncChangesToServer();
+                    final Boolean newValue = !typedSetting.getValue(blueprintRenderSettings);
+                    typedSetting.setValue(blueprintRenderSettings, newValue);
+                    buttonImage.setText(Component.translatable(newValue ? "options.on" : "options.off"));
+
+                    // TODO: ray why? RenderingCache.getOrCreateBlueprintPreviewData(bluePrintId).syncChangesToServer();
```

Comment:
@Raycoms

Reply:
The moment you turn this on, its not instantly shared if you dont do this. It would then only be shared on a bigger blueprint change

Reply:
Oh,kk, so it's basically config watcher thing, will move